docs/ota-client-guide/modules/ROOT/pages/hsm-provisioning-example.adoc

```diff
@@ -182,7 +183,24 @@ key_source = "pkcs11"
 base_path = "/var/sota/import"
 tls_cacert_path = "root.crt"
 ----
++
+Note: on Ubuntu Bionic, the OpenSC pkcs11 module lies in `/usr/lib/x86_64-linux-gnu/pkcs11/opensc-pkcs11.so`.
 
 == Provisioning the device
 
 If all these steps have been followed, the device will establish a TLS connection to the backend using the HSM and will sign its manifests with the uptane key.
+
+== Yocto integration
+
+You can here refer to the instructions in xref:enable-device-cred-provisioning.adoc[Enable device-credential provisioning and install device certificates] but use OpenSC instead of SoftHSM:
+
+----
+IMAGE_INSTALL_append = " opensc"
+SOTA_CLIENT_FEATURES = "hsm"
+SOTA_CLIENT_PROV = "aktualizr-device-prov-hsm"
+SOTA_DEPLOY_CREDENTIALS = "0"
+----
+
+Also, the configuration fragment `/usr/lib/sota/conf.d/20-sota-device-cred-hsm.toml` will also have to be modified, as detailed in the previous section, for example with a `.bbappend`.
+
+Note that for the moment, the gateway url and root certificate will still need to be copied manually to the device.
```

Comment:
Is this a hard limitation for some reason, or could we expect that someone using this in production would be able to write (or modify) a recipe to include these in the build?

Reply:
There is no technical obstacle as far as I know. And, indeed, integrators will have to choose how to pass that down to the devices. Putting it inside the image directly should also be fine as it's not secret and not device-specific.